raiden/exceptions.py

```diff
@@ -41,7 +49,7 @@ class DepositOverLimit(RaidenError):
     pass
 
 
-class DepositMismatch(RaidenError):
+class DepositMismatch(RaidenUnrecoverableError):
```

Comment:
The psudo code you've provided is actually implemented. However, we do not get `total_deposit` from the contracts but rather, only the `deposit` amount that the current node has contributed.
With that being said, if the transaction fails, we re-fetch the deposit amount from the contract and compare it with the deposit amount the call had. 
https://github.com/raiden-network/raiden/pull/2168/files#diff-d02574a7a7eca826412780e80a6326a3R1094
and check if something changed... otherwise, we raise the unrecoverable exception.